vignettes/xcms.Rmd

```diff
@@ -1106,6 +1106,110 @@ properties of the mice analyzed (sex, age, litter mates etc).
 
+## Quality-based filtering of features
+
+When dealing with metabolomics results, it is often necessary to filter
+features based on certain criteria.
+These criteria are typically derived from statistical formulas applied to full
+rows of data, where each row represents a feature. The `filterFeatures`
+function provides a robust solution for filtering features based on these
+conventional quality assessment criteria. It supports multiple types of
+filtering, allowing users to tailor the filtering process to their specific
+needs, all controlled by the `filter` argument. This function and its
+implementations are applicable to both `XcmsExperiment` results objects and
+`SummarizedExperiment` objects.
+
+We will demonstrate how to use the `filterFeatures` function to perform quality
+assessment and filtering on both the faahko and res objects. The `filter`
+argument can accommodate various types of input, each determining the
+specific type of quality assessment and filtering to be performed.
+
+The `RsdFilter` enable users to filter features based on their coefficient of
+variation for a specified `threshold`. It is recommended to base the
+computation on quality control (QC) samples, as demonstrated below:
+
+```{r}
+# Set up parameters for RsdFilter
+rsd_filter <- RsdFilter(threshold = 0.3,
+                        qcIndex = sampleData(faahko)$sample_type == "QC")
+
+# Apply the filter to faakho object
+filtered_faahko <- filterFeatures(object = faahko, filter = rsd_filter)
+
+# Now apply the same strategy to the res object
+rsd_filter <- RsdFilter(threshold = 0.3, qcIndex = res$sample_type == "QC")
+filtered_res <- filterFeatures(object = res, filter = rsd_filter, assay = "raw")
+```
+
+The `DratioFilter`  can be used to filter features based on the D-ratio or
+*dispersion ratio*, which compares the standard deviation in QC samples to that
+in QC samples.
+
+```{r}
+# Set up parameters for DratioFilter
+dratio_filter <- DratioFilter(
+    threshold = 0.5,
+    qcIndex = sampleData(filtered_faahko)$sample_type == "QC",
+    studyIndex = sampleData(filtered_faahko)$sample_type == "study")
+
+# Apply the filter to faahko object
+filtered_faakho <- filterFeatures(object = filtered_faahko,
+                                  filter = dratio_filter)
+
+# Now same but for the res object
+dratio_filter <- DratioFilter(
+    threshold = 0.5,
+    qcIndex = filtered_res$sample_type == "QC",
+    studyIndex = filtered_res$sample_type == "study")
+
+filtered_res <- filterFeatures(object = filtered_res,
+                               filter = dratio_filter)
+```
+
+The `PercentMissingFilter` allows to filter features based on the percentage of
+missing values for each feature. This function takes as an input the parameter
+`f` which is supposed to be a vector of the length of the object with the
+sample type for each samples. The function then compute the percentage of
+missing values per sample groups and filters features based on this. Another
```

Comment:
Maybe also specify which features get removed:

*based on this. Features with a percent of missing values larger than `threshold` in **all** sample groups will be removed. ...*